lldb/include/lldb/Core/Debugger.h

```diff
@@ -569,6 +569,18 @@ class Debugger : public std::enable_shared_from_this<Debugger>,
   SetDestroyCallback(lldb_private::DebuggerDestroyCallback destroy_callback,
                      void *baton);
 
+  /// Add a notification callback when notification type event happens. Return a
```

Comment:
Read literally this says you are adding the notification callback when the event happens.  Maybe:

Add a notification callback that will trigger when an event of the given NotificationType occurs.

You should also state when the creation and destruction callbacks trigger w.r.t. the action the event is reporting.  I.e. Creation callbacks trigger synchronously after the given object is fully set up, and Destruction callbacks trigger right before the object is destroyed.